src/cli/workflows.ts

```diff
@@ -183,8 +185,11 @@ export const wf = () => {
         opts.activate
       ];
       logOp(cmd, args);
+      if (opts.deactivateBefore) {
+        await wf.deactivate(wfFilter); // Deactivate to avoid conflict
+      }
+      // Proceed with publishing
       if (opts.dry === false) {
-        await wf.deactivate(wfFilter) // Deactivate to avoid conflict
         await wf.publish(...args);
       }
```

Comment:
Actually it was correct to place deactivate inside that `if`. We making changes only if it is not dry run. Same below.